src/app/core/guards/login.guard.ts

```diff
@@ -0,0 +1,46 @@
+import { Injectable } from '@angular/core';
+import { ActivatedRouteSnapshot, CanActivate, NavigationEnd, Router, RouterStateSnapshot } from '@angular/router';
+import { NgbModal, NgbModalRef } from '@ng-bootstrap/ng-bootstrap';
+import { Store, select } from '@ngrx/store';
+import { filter, first } from 'rxjs/operators';
+
+import { getUserAuthorized } from 'ish-core/store/user';
+import { whenTruthy } from 'ish-core/utils/operators';
+import { LazyLoginModalComponent } from 'ish-shell/header/components/lazy-login-modal/lazy-login-modal.component';
+
+/**
+ * triggers logging out the user if the guarded route is visited
+ */
+@Injectable({ providedIn: 'root' })
+export class LoginGuard implements CanActivate {
+  currentDialog: NgbModalRef<typeof LazyLoginModalComponent>;
+
+  constructor(private modalService: NgbModal, private router: Router, private store: Store<{}>) {}
+
+  canActivate(route: ActivatedRouteSnapshot, _: RouterStateSnapshot) {
+    this.currentDialog = this.modalService.open(LazyLoginModalComponent, { centered: true });
+    this.currentDialog.componentInstance.loginMessageKey = route.queryParamMap.get('messageKey');
+
+    this.router.events
+      .pipe(
+        filter(event => event instanceof NavigationEnd),
+        first()
+      )
+      .subscribe(() => {
+        this.currentDialog.close();
+      });
+    this.store
+      .pipe(
+        select(getUserAuthorized),
+        whenTruthy(),
+        first()
+      )
+      .subscribe(() => this.currentDialog.close());
+
+    // Go back to home page after the modal is closed
+    this.currentDialog.result.then(() => {
```

Comment:
Suggestion:

```ts
return from(this.currentDialog.result).pipe(
  mapTo(this.router.createUrlTree(route.queryParams.returnUrl || '/home'))
);

Reply:
I'd say the return always has to be false, so the routing gets cancelled.